### PR TITLE
Add Items Index Functionality 

### PR DESCRIPTION
**Pull request by lpile**
Add Items Index Functionality 
--
* Add instance methods #activate and #deactivate specs
* Add instance methods #activate and #deactivate
* Add class method .all_active spec
* Add class method .all_active
* Add displays all active items spec
* Update #index action
* Add displays specific attributes spec
* Add link_to item name to item show page spec
* Add users#show route
* Add link_to item image to item show page spec
* Add #show action
* Add edge case specs
* Update template to pass spec
* All tests passing

### CODE REVIEW

File: app/views/items/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<% if @items.empty? %>
+  <h1>There are no items for sell.</h1>
+<% else %>
```
Comment (Martsy) [inline app/views/items/index.html.erb]: Can you change sell to sale?

File: spec/features/items/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+      expect(page).to have_content(@item_1.name)
+      expect(page).to have_content(@item_1.description)
+      expect(page).to have_css("img[src='#{@item_1.image}']")
+      expect(page).to have_content(@item_1.user.name)
+      expect(page).to have_content(@item_1.inventory)
+      expect(page).to have_content(@item_1.price)
+    end
+
+    it "it displays average amount of time it takes merchant to fulfill item"
```
Comment (DavisC0801) [inline spec/features/items/show_spec.rb]:
```suggestion
    xit "it displays average amount of time it takes merchant to fulfill item"
```
Using the skip function is a better #todo flag for me personally.

File: spec/features/items/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    it "it displays a link to add this item to cart"
```
Comment (DavisC0801) [inline spec/features/items/show_spec.rb]:
```suggestion
    xit "it displays a link to add this item to cart"
```
